### DISCUSSION TITLE
How do transform message for unhandled exceptions

### DISCUSSION THREAD

**Discussion by MeikelLP** [xunit/xunit#2997] [Question] [answered]
While writing tests my code may throw a `Refit.ValidationApiException`. Sadly their message is not very helpful "Response status code does not indicate success: 400 (Bad Request)." and I'd like to adjust the exception message to also log the response body.

How can I archive this?

**Comment by bradwilson** [comment, marked as answer]
There is no hook for you to do this inside of xUnit.net. You'll have to add a `try`/`catch` block around your test code to catch that specific exception, and then throw one with a better message.
